### PR TITLE
Fix Bayonets + Faster Attachment

### PR DESCRIPTION
**Pull request by Surrealaser**
Bayonets now attach in 1 second if there is no attachment to remove on the muzzle slot. Fix Bayonets!

All other attachments attach 3 seconds faster (3 seconds, down from 6) because fuck waiting for so long for no reason.

EDIT: If you're untrained with firearms, removing/adding attachments takes twice as long and you look like a bumbling idiot.

### CODE REVIEW

**Comment by CosmicScientist** [comment]
It might be good to change detachment too.

**Comment by Surrealaser** [comment]
I did

**Comment by CosmicScientist** [comment]
👍

**Comment by ninjanomnom** [comment]
I'm not against improved attachments speed but this seems a good candidate to having it be based on the firearm skill

**Comment by MoXuT0** [comment]
After some testing deattachment removal is instant and attachment is still 6 seconds.
Deattachment can probably be fixed with `if(!do_after(usr,detach_delay, TRUE, 5, BUSY_ICON_FRIENDLY))
return` in 515. Will continue to test more.

**Comment by Surrealaser** [comment]
@MoXuT0 Fixed everything and added some finishing touches.

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 1]
```diff
@@ -342,9 +342,26 @@ should be alright.
...
+			"<span class='notice'>You fix [attachment]!</span>", null, 4)
+		else
+			user.visible_message("<span class='notice'>[user] begins attaching [attachment] to [src].</span>",
+			"<span class='notice'>You begin attaching [attachment] to [src].</span>", null, 4)
+		if(firearm_skill >= 2) //See if the attacher is super skilled/panzerelite born to defeat never retreat etc
+			attach_delay *= 0.5		
+	else //If the user has no training, attaching takes twice as long and they fumble about, looking like a retard.
+		attach_delay *= 2
+		user.visible_message("<span class='notice'>[user] begins fumbling about, trying to attach [attachment] to [src].</span>",
+		"<span class='notice'>You begin funmbling about, trying to attach [attachment] to [src].</span>", null, 4)
```
Comment (MoXuT0) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: Funmbling? Pretty sure you meant fumbling. Same thing in 525.

**Comment by Surrealaser** [comment]
@MoXuT0 Yes, they're typos; probably hit n at the same time I hit m. Anyways, fixed.

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 2]
```diff
@@ -342,9 +342,26 @@ should be alright.
...
 		return
 
-	user.visible_message("<span class='notice'>[user] begins attaching [attachment] to [src].</span>",
-	"<span class='notice'>You begin attaching [attachment] to [src].</span>", null, 4)
-	if(do_after(user,60, TRUE, 5, BUSY_ICON_FRIENDLY))
+
+	var/attach_delay = 30
+	var/firearm_skill = user.mind.cm_skills.firearms
+	if(attachment.name == "bayonet" && (!src.muzzle || src.muzzle.name == "bayonet")) //Attach  bayonet fast if no attachment to remove on the barrel, or you're replacing a bayo with another bayo
+		attach_delay = 10
```
Comment (ninjanomnom) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: If you want different delays based on attachment it needs to be an attachment var

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 3]
```diff
@@ -342,9 +342,26 @@ should be alright.
...
+	if (user.mind && user.mind.cm_skills && firearm_skill)
+		if(attachment.name == "bayonet")
+			user.visible_message("<span class='notice'>[user] fixes [attachment]!.</span>",
+			"<span class='notice'>You fix [attachment]!</span>", null, 4)
+		else
+			user.visible_message("<span class='notice'>[user] begins attaching [attachment] to [src].</span>",
+			"<span class='notice'>You begin attaching [attachment] to [src].</span>", null, 4)
+		if(firearm_skill >= 2) //See if the attacher is super skilled/panzerelite born to defeat never retreat etc
```
Comment (ninjanomnom) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: skill levels are defines, use them

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 4]
```diff
@@ -342,9 +342,26 @@ should be alright.
...
-	if(do_after(user,60, TRUE, 5, BUSY_ICON_FRIENDLY))
+
+	var/attach_delay = 30
+	var/firearm_skill = user.mind.cm_skills.firearms
+	if(attachment.name == "bayonet" && (!src.muzzle || src.muzzle.name == "bayonet")) //Attach  bayonet fast if no attachment to remove on the barrel, or you're replacing a bayo with another bayo
+		attach_delay = 10
+	if (user.mind && user.mind.cm_skills && firearm_skill)
+		if(attachment.name == "bayonet")
+			user.visible_message("<span class='notice'>[user] fixes [attachment]!.</span>",
+			"<span class='notice'>You fix [attachment]!</span>", null, 4)
```
Comment (ninjanomnom) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: Again if you want different attach texts it needs to be based on values inside the attachment

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 5]
```diff
@@ -493,10 +510,21 @@ should be alright.
...
 		return
 
-	usr.visible_message("<span class='notice'>[usr] begins stripping [A] from [src].</span>",
-	"<span class='notice'>You begin stripping [A] from [src].</span>", null, 4)
-
-	if(!do_after(usr,35, TRUE, 5, BUSY_ICON_FRIENDLY))
+	var/detach_delay = 30
+	var/firearm_skill = usr.mind.cm_skills.firearms
+	if(A.name == "bayonet") //Detach the bayonet fast
+		detach_delay = 10
```
Comment (ninjanomnom) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: attachment var

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 6]
```diff
@@ -493,10 +510,21 @@ should be alright.
...
+	if (usr.mind && usr.mind.cm_skills && firearm_skill)
+		usr.visible_message("<span class='notice'>[usr] begins stripping [A] from [src].</span>",
+		"<span class='notice'>You begin stripping [A] from [src].</span>", null, 4)
+		if(firearm_skill >= 2) //See if the attacher is super skilled/panzerelite born to defeat never retreat etc
```
Comment (ninjanomnom) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: skill define

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 7]
```diff
@@ -342,9 +342,18 @@ should be alright.
 		to_chat(user, "<span class='warning'>The attachment on [src]'s [attachment.slot] cannot be removed!</span>")
 		return
 
-	user.visible_message("<span class='notice'>[user] begins attaching [attachment] to [src].</span>",
-	"<span class='notice'>You begin attaching [attachment] to [src].</span>", null, 4)
-	if(do_after(user,60, TRUE, 5, BUSY_ICON_FRIENDLY))
+	var/final_delay = attachment.attach_delay
+	if (usr.mind.cm_skills.firearms)
```
Comment (ninjanomnom) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: `usr` should not be used except in direct access verbs, switch this to `user`

File: code/modules/projectiles/updated_projectiles/gun_helpers.dm [hunk 8]
```diff
@@ -342,9 +342,18 @@ should be alright.
...
+		user.visible_message("<span class='notice'>[user] begins attaching [attachment] to [src].</span>",
+		"<span class='notice'>You begin attaching [attachment] to [src].</span>", null, 4)
+		if(usr.mind.cm_skills.firearms >= SKILL_FIREARMS_DEFAULT) //See if the attacher is super skilled/panzerelite born to defeat never retreat etc
```
Comment (ninjanomnom) [inline code/modules/projectiles/updated_projectiles/gun_helpers.dm]: same as above

**Comment by Rohesie** [comment]
The balance impact is really small and it makes the game experience more enjoyable. More of a QoL change. I'm merging this.

**Pull request merged**